packages/e2e-tests/specs/editor/blocks/code.test.js

```diff
@@ -20,4 +21,15 @@ describe( 'Code', () => {
 
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should delete block when backspace in an empty code', async () => {
+		await insertBlock( 'Code' );
+		await page.keyboard.type( 'a' );
```

Comment:
Nit pick: this is not really necessary to test the behaviour. I try to make e2e tests as small as small as they can be.